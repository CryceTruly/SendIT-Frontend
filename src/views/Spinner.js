import React from "react";

const Spinner = props => (
  <div className="d-flex text-primary justify-content-center">
    <div className="spinner-border" role="status">
      <span className="sr-only">Loading...</span>
    </div>
  </div>
);

export default Spinner;
