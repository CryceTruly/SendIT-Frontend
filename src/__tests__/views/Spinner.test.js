import React from "react";
import { shallow } from "enzyme";
import Spinner from "../../views/Spinner";

describe("Spinner tests", () => {
  it("should not regress", () => {
    const componentWrapper = shallow(<Spinner />);
    expect(componentWrapper).toMatchSnapshot();
  });
});
