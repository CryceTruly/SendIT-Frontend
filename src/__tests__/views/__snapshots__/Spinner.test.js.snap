// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="spinner-border"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>,
      "className": "d-flex text-primary justify-content-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="sr-only"
        >
          Loading...
        </span>,
        "className": "spinner-border",
        "role": "status",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "sr-only",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "span",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="spinner-border"
          role="status"
        >
          <span
            className="sr-only"
          >
            Loading...
          </span>
        </div>,
        "className": "d-flex text-primary justify-content-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="sr-only"
          >
            Loading...
          </span>,
          "className": "spinner-border",
          "role": "status",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "sr-only",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "span",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
